ext/rotoscope/rotoscope.c

```diff
@@ -13,11 +13,17 @@
 #include "strmemo.h"
 #include "tracepoint.h"
 
-VALUE cRotoscope, cTracePoint;
+VALUE cRotoscope, cTracePoint, cThread;
+ID currentId;
 
 // recursive with singleton2str
 static rs_class_desc_t class2str(VALUE klass);
 
+static unsigned long gettid() {
+  VALUE cur_thread = rb_funcall(cThread, currentId, 0);
```

Comment:
huh, I hunted through the codebase and couldn't find a convenience method for it! I assumed it had to exist. Thanks for finding it!